### PR TITLE
#78 Define interface for the self-defined Mover

### PR DESCRIPTION
**Pull request by timmyyao**

### CODE REVIEW

File: hadoop-ssm-project/src/main/java/org/apache/hadoop/ssm/mover/MoverPool.java [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+  public static void init(Configuration configuration) {
+    conf = configuration;
+  }
+
+  /**
+   * Create a Mover event.
+   * @param dir the directory to enforce the storage policy using Mover tool.
+   * @return the UUID of this action for user to track status
+   */
+  public UUID createMoverAction(String dir) {
```
Comment (littlezhou) [inline hadoop-ssm-project/src/main/java/org/apache/hadoop/ssm/mover/MoverPool.java]: just a reminder, an move action can also be applied on a single file, so parameter 'dir' may cause some misunderstanding.

File: hadoop-ssm-project/src/main/java/org/apache/hadoop/ssm/mover/MoverPool.java [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+  public Status getStatus(UUID id) {
+    return moverMap.get(id);
+  }
+
+  /**
+   * Remove the status of a certain moving event.
+   * After removing, the status of this event can no longer be tracked.
+   * @param id the UUID of the event
+   */
+  public void removeStatus(UUID id) {
```
Comment (littlezhou) [inline hadoop-ssm-project/src/main/java/org/apache/hadoop/ssm/mover/MoverPool.java]: who will call this function? when?

File: hadoop-ssm-project/src/main/java/org/apache/hadoop/ssm/mover/MoverStatus.java [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+
+/**
+ * Status of Mover tool.
+ */
+public class MoverStatus extends Status {
+  private Boolean isFinished = false;
+  private long startTime = 0;
+  private Boolean succeeded = false;
+  private long totalDuration = 0;
+
```
Comment (littlezhou) [inline hadoop-ssm-project/src/main/java/org/apache/hadoop/ssm/mover/MoverStatus.java]: it's good enough for starting, but we'd better to provide more insightful metrics later.

File: hadoop-ssm-project/src/main/java/org/apache/hadoop/ssm/mover/MoverPool.java [hunk 4]
```diff
@@ -0,0 +1,113 @@
+/**
```
Comment (littlezhou) [inline hadoop-ssm-project/src/main/java/org/apache/hadoop/ssm/mover/MoverPool.java]: it seems no persistence mechanism implemented.

File: hadoop-ssm-project/src/main/java/org/apache/hadoop/ssm/mover/MoverStatus.java (hunk 3 above)
Comment (timmyyao) [inline hadoop-ssm-project/src/main/java/org/apache/hadoop/ssm/mover/MoverStatus.java]: I have created another issue to improve the status and control of Mover event.

File: hadoop-ssm-project/src/main/java/org/apache/hadoop/ssm/mover/MoverPool.java (hunk 1 above)
Comment (timmyyao) [inline hadoop-ssm-project/src/main/java/org/apache/hadoop/ssm/mover/MoverPool.java]: I have changed it to "path"

File: hadoop-ssm-project/src/main/java/org/apache/hadoop/ssm/mover/MoverPool.java (hunk 2 above)
Comment (timmyyao) [inline hadoop-ssm-project/src/main/java/org/apache/hadoop/ssm/mover/MoverPool.java]: User should call this function. I have written into comment.

File: hadoop-ssm-project/src/main/java/org/apache/hadoop/ssm/mover/MoverPool.java (hunk 4 above)
Comment (timmyyao) [inline hadoop-ssm-project/src/main/java/org/apache/hadoop/ssm/mover/MoverPool.java]: Would you mind me doing it in a new issue?

**Pull request merged**